tools/c7n_gcp/c7n_gcp/filters/metrics.py

```diff
@@ -195,6 +197,8 @@ def split_by_resource(self, metric_list):
     def process_resource(self, resource):
         resource_metric = resource.setdefault('c7n.metrics', {})
         resource_name = self.manager.resource_type.get_metric_resource_name(resource)
+        if self.metric_key == 'resource.labels.instance_id':
```

Comment:
it does feel like we should be able to defer this compute specific key and id back to the resource, it does appear gcp metrics do have some variance against the id field they tracked, but hardcoding in compute specific behavior into a base class makes it harder to handle. so it seems to me a better option would be to  have the delegation from this base class to resource type specific identification of id based on metric, rather than override name in the base class for a specific use case.